src/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -11,16 +11,20 @@ namespace System.Net.Http
 {
     public class StreamContent : HttpContent
     {
-        private const int DefaultBufferSize = 4096;
-
         private Stream _content;
         private int _bufferSize;
         private bool _contentConsumed;
         private long _start;
 
         public StreamContent(Stream content)
-            : this(content, DefaultBufferSize)
         {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            }
+
```

Comment:
While we are discussing this particular topic (preserving compat on error exceptions), do we have a test for that?  It sounds like we don't since if we did, it would have failed (or need to be changed).

Reply:
It looks like we have tests for each failure individually, but not together.
https://github.com/dotnet/corefx/blob/c351ba1f094871e4ae97f23f20f26648662cc4e6/src/System.Net.Http/tests/FunctionalTests/StreamContentTest.cs#L23-L33

Reply:
Free free to add another test to provide coverage for this change. :)